python/distant_vfx/colour/utilities.py

```diff
@@ -0,0 +1,189 @@
+            root_dir=scan_folder
+        )
+        if cube_path:
+            return cube_path
+    for descriptor in scans:
+        cube_path = search_for_cube_by_descriptor(
+            descriptor=descriptor,
+            plates=scans,
+            root_dir=scan_folder
+        )
+        if cube_path:
+            return cube_path
+    return None
+
+
+def _get_proj_seq_shot_descriptor_version(path):
+    basename = os.path.basename(path).replace(".", "_")
+    splits = basename.split("_")
+    version_splits = [
+        split[1:]
+        for split in splits
+        if split.startswith("v") and split[1:].isdigit()
+    ]
+
+    return (
+        "dst",
+        basename[:3].lower(),
+        basename[:7].lower(),
+        splits[1].lower() if len(splits) > 1 else "",
+        version_splits[0] if len(version_splits) else "001"
+    )
+
+
+def descriptor_from_filename(filename):
+    project, sequence, shot, descriptor, version = _get_proj_seq_shot_descriptor_version(filename)
+
+    try:
+        number_version = int(version)
+    except:
+        number_version = 1
+    data_struct = {
+        "project": project,
+        "sequence": sequence,
+        "shot": shot,
+        "descriptor": descriptor,
+        "version": number_version,
+        "filename": filename,
+    }
+
+    return data_struct
+
+
+def search_for_cube_by_descriptor(descriptor, plates, root_dir):
+    # Use the bg first
+    background_versions = plates.get(descriptor, [])
+    for version_num in sorted(background_versions.keys(), reverse=True):
+        data_struct = background_versions[version_num]
+
+        cube_path = os.path.join(
+            root_dir,
+            data_struct.get("filename"),
+            "_cdl",
+            "versioned",
+            data_struct.get("filename"),
+            "luts",
+            data_struct.get("filename") + ".cube"
+        )
+        if os.path.exists(cube_path):
+            # Perfect! Use it!
+            return cube_path
+    return None
+
+
+def _path_get_shot_lut_from_path(path):
+    "/mnt/Projects/dst/post/shot/vpd/vpd0070/shot/vpd0070_comp_v003/2156x1500_exr/vpd0070_comp_v003.%04d.exr"
+    "/mnt/Projects/dst/post/shot/ftc/ftc0200"
+    "/mnt/Projects/dst/post/scan/ftc1000_bg_v001/_cdl/versioned/ftc1000_bg_v001/luts/ftc1000_bg_v001.cube"
+
+    "/mnt/Projects/dst/post/shot/ftc/ftc1000/shot/ftc1000_comp_v001/2156x1500_exr/ftc1000_comp_v001.%04d.exr"
+    "/mnt/Projects/dst/post/shot/ftc/ftc1000/plate/ftc1000_bg_v001/_cdl/versioned/ftc1000_bg_v001/luts/ftc1000_bg_v001.cube"
+    dirname_three = os.path.dirname(os.path.dirname(os.path.dirname(path)))
+    version_folder_name = os.path.basename(os.path.dirname(os.path.dirname(path)))
+    if os.path.basename(dirname_three) == "shot":
+        # We're dealing with a rendered image!
+        shot_dir = os.path.dirname(os.path.dirname(dirname_three))
+        plates_dir = os.path.join(shot_dir, "plate")
+        plates = {}
+        for folder_name in plates_dir:
```

Comment:
I may be missing something, but should this be:
`for folder_name in os.listdir(plates_dir):`? Seems like `plates_dir` is a path.

Reply:
Yes you are correct! Wrote it sorta late.

Reply:
Updated